imports/ui/components/common/Nestable.jsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react'
+
+export default class Nestable extends React.Component{
+  render(){
+    var items = this.props.items;
+
+    return(
+      <div className="dd" id="nestable">
+        <OrderedList items={items}/>
+      </div>
+    );
+  }
+}
+
+class Item extends Nestable{
+  render(){
+    return(
+      <li className="dd-item" >
+        <div className="dd-handle" style={styles.item}>{this.props.itemName}</div>
+      </li>
+    );
+  }
+}
+
+class ListItem extends Nestable{
+  constructor(){
+    super();
+    this.state = {
+      isCollapsed: true
+    }
+  }
+
+  toggleCollapse() {
+      this.setState({isCollapsed: !this.state.isCollapsed});
+  }
+  render(){
+    var isCollapsed = this.state.isCollapsed;
+    var listName = this.props.listName;
+
+    return(
+      <li className="dd-item">
+        <div  className="dd-handle" style={styles.list}
+              onClick={this.toggleCollapse.bind(this)}>
+          <i  className={"fa fa-"+ (isCollapsed ? "plus" : "minus")}
+              style={styles.collapseIcon}>
+          </i>
+          {listName}
+        </div>
+        <OrderedList items={this.props.items} isCollapsed={isCollapsed}/>
+      </li>
+    );
+  }
+}
+
+class OrderedList extends Nestable{
```

Comment:
Interesting way of structuring if else statements within React components! The flow of logic is clear and I know what gets rendered / not rendered. I think I will try to follow your style when I'm dealing with React components which hold a lot of state information.  👍